openedx/core/djangoapps/xmodule_django/models.py

```diff
@@ -181,6 +183,19 @@ class UsageKeyField(OpaqueKeyField):
     KEY_CLASS = UsageKey
 
 
+class UsageKeyWithRunField(UsageKeyField):
```

Comment:
I agree with your choice of option 1 here. It allows more flexibility for this data model over the longer term for any sort of published XBlock content, regardless of whether it's really a "course", a subset of a course, or something else entirely.